Modules/Core/Common/include/itkMetaDataObject.hxx

```diff
@@ -28,9 +28,79 @@
 #ifndef itkMetaDataObject_hxx
 #define itkMetaDataObject_hxx
 
+#include <type_traits>
+#include <iterator>
+
+template <typename T, typename = void>
+inline constexpr bool is_iterable_print_v = false;
+// Specialize for std::vector<T> and std::vector<std::vector<T>>
+template <typename T>
+inline constexpr bool is_iterable_print_v<std::vector<T>, std::void_t<>> = true;
+template <typename T>
+inline constexpr bool is_iterable_print_v<std::vector<std::vector<T>>, std::void_t<>> = true;
```

Comment:
Nice to see a use case for variable templates, thanks @phcerdan. Unfortunately "itkMetaDataObject.hxx" is usually `#include`'d (indirectly) by end users, and I guess it's not the intention to add `is_iterable_print_v` to the global namespace of the user, right? If you agree, you may possibly make `is_iterable_print_v` a private static member of `MetaDataObject`. Or otherwise, maybe it's simpler to just add overloaded private static `MetaDataObject::IsIterablePrint(const T&)` member functions. What do you think?

Reply:
Good points @N-Dekker, I will make static private members

Reply:
GCC doesn't like it... (moving the to static inline inside the class).
https://stackoverflow.com/questions/72190700/explicit-template-argument-list-not-allowed-with-g-but-compiles-with-clang

I will explore further.

Reply:
It seems fixed in gcc14, but we cannot afford that.

Reply:
Another option I guess would be to have constexpr static member functions `IsIterablePrint(const T&)` overloaded (rather than "specialized") for `std::vector<T>` and `std::vector<std::vector<T>>`, right?

Reply:
Regarding the name of the boolean, maybe something like `ShouldBePrintedElementWise` ?